x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/SparseVectorQueryBuilder.java

```diff
@@ -0,0 +1,367 @@
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.inference.InferenceResults;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.ml.action.CoordinatedInferenceAction;
+import org.elasticsearch.xpack.core.ml.action.InferModelAction;
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelPrefixStrings;
+import org.elasticsearch.xpack.core.ml.inference.results.TextExpansionResults;
+import org.elasticsearch.xpack.core.ml.inference.results.WarningInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.TextExpansionConfigUpdate;
+import org.elasticsearch.xpack.core.ml.search.TokenPruningConfig;
+import org.elasticsearch.xpack.core.ml.search.WeightedToken;
+import org.elasticsearch.xpack.core.ml.search.WeightedTokensUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+import static org.elasticsearch.xpack.core.ClientHelper.ML_ORIGIN;
+import static org.elasticsearch.xpack.core.ClientHelper.executeAsyncWithOrigin;
+
+public class SparseVectorQueryBuilder extends AbstractQueryBuilder<SparseVectorQueryBuilder> {
+    public static final String NAME = "sparse_vector";
+    public static final String ALLOWED_FIELD_TYPE = "sparse_vector";
+    public static final ParseField FIELD_FIELD = new ParseField("field");
+    public static final ParseField QUERY_VECTOR_FIELD = new ParseField("query_vector");
+    public static final ParseField INFERENCE_ID_FIELD = new ParseField("inference_id");
+    public static final ParseField QUERY_FIELD = new ParseField("query");
+    public static final ParseField PRUNE_FIELD = new ParseField("prune");
+    public static final ParseField PRUNING_CONFIG_FIELD = new ParseField("pruning_config");
+
+    private static final boolean DEFAULT_PRUNE = false;
+
+    private final String fieldName;
+    private final List<WeightedToken> queryVectors;
+    private final String inferenceId;
+    private final String query;
+    private final boolean shouldPruneTokens;
+
+    private final SetOnce<TextExpansionResults> weightedTokensSupplier;
+
+    @Nullable
+    private final TokenPruningConfig tokenPruningConfig;
+
+    public SparseVectorQueryBuilder(String fieldName, String inferenceId, String query) {
+        this(fieldName, null, inferenceId, query, DEFAULT_PRUNE, null);
+    }
+
+    public SparseVectorQueryBuilder(String fieldName, List<WeightedToken> queryVector) {
```

Comment:
Looks like this ctor is not used anywhere?

Reply:
Removed